spark/src/main/java/org/apache/iceberg/actions/RewriteManifestsAction.java

```diff
@@ -162,8 +169,6 @@ public RewriteManifestsAction useCaching(boolean newUseCaching) {
 
   @Override
   public RewriteManifestsActionResult execute() {
-    Preconditions.checkArgument(stagingLocation != null, "Staging location must be set");
```

Comment:
What if we kept this and did a check whether the table implements `HasTableOperations`? Then we wouldn't need an unchecked cast.

Reply:
We could but we use the same principle in a number of places, including the remove orphan files action.

Reply:
Instead of relying on `TableOperations`, we could check table properties directly too.

Reply:
My worry is that it may be confusing that we default the staging location only in some cases.